### PR TITLE
Aggregations Refactor: Refactor Nested and Reverse Nested Aggregations

### PR DESCRIPTION
**Pull request by colings86**

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/search/aggregations/bucket/nested/NestedAggregator.java [hunk 1]
```diff
@@ -139,13 +146,28 @@ private static Query findClosestNestedPath(Aggregator parent) {
...
 
-        public Factory(String name, String path) {
+        public Factory(String name) {
             super(name, InternalNested.TYPE);
+        }
+
+        /**
+         * Set the path to use for this nested aggregation. The path must match
+         * the path to a nested object in the mappings. This parameter is
+         * compulsory.
```
Comment (cbuescher) [inline core/src/main/java/org/elasticsearch/search/aggregations/bucket/nested/NestedAggregator.java]: I'm curious why you moved `path` aways from the constructor if it is compulsory?

Comment (colings86) [inline core/src/main/java/org/elasticsearch/search/aggregations/bucket/nested/NestedAggregator.java]: Good point, I'll move it back to the constructor.

**Comment by cbuescher** [comment]
Looks good to me, left one question but maybe @jpountz should also have a final look.

**Comment by jpountz** [comment]
LGTM

**Pull request merged**